webhooks/virtualmachinesnapshot/validation/virtualmachinesnapshot_validator.go

```diff
@@ -56,22 +61,85 @@ func (v validator) For() schema.GroupVersionKind {
 	return vmopv1.GroupVersion.WithKind(reflect.TypeOf(vmopv1.VirtualMachineSnapshot{}).Name())
 }
 
+// ValidateCreate makes sure the VirtualMachineSnapshot create request is valid.
 func (v validator) ValidateCreate(ctx *pkgctx.WebhookRequestContext) admission.Response {
-	// TODO: Implement validation for create
+	vmSnapshot, err := v.vmSnapshotFromUnstructured(ctx.Obj)
+	if err != nil {
+		return webhook.Errored(http.StatusBadRequest, err)
+	}
 
 	validationErrs := make([]string, 0)
 
+	if vmSnapshot.Spec.VMRef == nil {
+		validationErrs = append(validationErrs, "spec.vmRef must not be nil")
+	}
+
+	if vmSnapshot.Spec.VMRef != nil {
+		if vmSnapshot.Spec.VMRef.Kind != virtualMachineKind {
+			validationErrs = append(validationErrs, fmt.Sprintf("spec.vmRef.kind must be %q, got %q",
```

Comment:
PTAL at our other validation webhooks like the virtualmachine one, use the `field` package to accumulate errors in a `field.ErrorList`.